scripts/o3de/ExportScripts/export_source_android.py

```diff
@@ -0,0 +1,218 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import o3de.command_utils as command_utils
+import o3de.export_project as exp
+import export_utility as eutil
+import o3de.manifest as manifest
+
+from o3de import android, android_support
+
+
+import argparse
+import logging
+import pathlib
+import sys
+
+from typing import List
+
+ASSET_PLATFORM = 'android'
+def export_source_android_project(ctx: exp.O3DEScriptExportContext,
+                           target_android_project_path: pathlib.Path,
+                           seedlist_paths: List[pathlib.Path],
+                           seedfile_paths: List[pathlib.Path],
+                           level_names: List[str],
+                           asset_pack_mode: str,
+                           engine_centric: bool,
+                           should_build_tools: bool,
+                           should_build_all_assets: bool,
+                           build_config: str,
+                           tool_config: str,
+                           tools_build_path: pathlib.Path | None,
+                           max_bundle_size: int,
+                           fail_on_asset_errors: bool,
+                           deploy_to_device: bool,
+                           org_name: str|None,
+                           activity_name:str|None,
+                           logger: logging.Logger|None) -> None:
+    o3de_cli_script_path = ctx.engine_path / ('scripts/o3de' + android_support.O3DE_SCRIPT_EXTENSION)
+
+    if not logger:
+        logger = logging.getLogger()
+        logger.setLevel(logging.ERROR)
+
+    is_installer_sdk = manifest.is_sdk_engine(engine_path=ctx.engine_path)
+
+    def run_android_command(args, cwd, error_msg):
+        has_errors = exp.process_command(args, cwd=cwd)
+        if has_errors:
+            raise exp.ExportProjectError(error_msg)
+    
+    run_android_command([o3de_cli_script_path, 'android-configure', '--validate'], cwd=ctx.project_path, error_msg="Invalid Android Configuration")
+
```

Comment:
I was hoping that instead of executing another process that invokes the python interpreter, that since we are in the same python project, we can run these commands using internal python calls and importing as necessary. For example, the equivalent of what you are trying to run above would be:

First create and initialize the same subprocessor as the android commands
```
    import o3de.android as android_processor

    # Create a subparser based on the android o3de commands
    arg_parser = argparse.ArgumentParser()
    android_subparser = arg_parser.add_subparsers(title="android sub-commands")
    android_processor.add_args(android_subparser)
```

And then for each time you run the 'android-configure'
```
    parsed_android_configure_command = arg_parser.parse_args(['android-configure', '--validate'])
    android_processor.configure_android_options(parsed_android_configure_command)
```